### PR TITLE
[Canvas] Add subtitles

### PR DESCRIPTION
**Pull request by RobinHoutevelts**
This PR adds the subtitle functionality of shows at canvas.be

### CODE REVIEW

File: youtube_dl/extractor/canvas.py [hunk 1]
```diff
@@ -53,6 +53,15 @@ def _real_extract(self, url):
                     'url': format_url,
                 })
         self._sort_formats(formats)
+        
+        subtitles = {}
+        for target in data['subtitleUrls']:
```
Comment (dstftw) [inline youtube_dl/extractor/canvas.py]: Extraction must tolerate missing `subtitleUrls`.

File: youtube_dl/extractor/canvas.py [hunk 2]
```diff
@@ -53,6 +53,15 @@ def _real_extract(self, url):
...
+            format_url, format_type = target.get('url'), target.get('type')
```
Comment (dstftw) [inline youtube_dl/extractor/canvas.py]: If there is no `url` you'll end up with invalid entry in `subtitles`.

File: youtube_dl/extractor/canvas.py (hunk 1 above)
Comment (RobinHoutevelts) [inline youtube_dl/extractor/canvas.py]: Even when it's _always_ there?
for example: https://mediazone.vrt.be/api/v1/canvas/assets/mz-ast-a30df0cf-590c-4c4f-b1fe-7c28fc3b7b8c

Comment (dstftw) [inline youtube_dl/extractor/canvas.py]: Today it's there - tomorrow it's not. Extractors must be as robust as possible and tolerate to layout changes. Subtitles is not a critical metadata to extract thus when `subtitleUrls` disappears (if ever) if must not break the main extraction routine.

Comment (RobinHoutevelts) [inline youtube_dl/extractor/canvas.py]: Seems unlikely but reasonable, I'll update and squash the commits tomorrow ( it's 5am, I'm off to bed )
Thanks reviewing the code

**Pull request closed without merging**